### PR TITLE
Astro 1160 checkbox

### PR DESCRIPTION
**Pull request by Cheshire89**
https://rocketcom.atlassian.net/browse/ASTRO-1160

update styling for astro checkbox
update checkbox page to checkbox group layout

### CODE REVIEW

File: static/css/astro.core.css [hunk 1]
```diff
@@ -1763,71 +1761,183 @@ label {
...
-  left: 5px;
-  height: 10px;
+.rux-checkbox--indeterminate input[type="checkbox"]:checked + label::after{
   width: 10px;
-  border-radius: 100%;
-  /* box-shadow: inset 0 0 1px 0 rgba(255, 255, 255, 0.9); */
-
-  background-color: var(--primary);
+  height: 5px;
+  transform: rotate(0deg);
```
Comment (dmcalester) [inline static/css/astro.core.css:1934]: What does `rotate(0deg)` accomplish here?

File: src/css/core/checkbox.css [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  border-right: 2px solid var(--controlTextColor);
+  border-top: 2px solid var(--controlTextColor);
+
+  transform: rotate(125deg);
+}
+
+.rux-checkbox--indeterminate input[type="checkbox"]:checked + label::after{
+  width: 10px;
+  height: 5px;
+  transform: rotate(0deg);
```
Comment (dmcalester) [inline src/css/core/checkbox.css:114]: What does `rotate(0deg)` accomplish here?

File: src/css/core/checkbox.css [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  position: absolute;
+  top: 5px;
+  display: flex;
+  content: "";
+}
+
+.rux-checkbox input[type="checkbox"]:checked + label::after{
+  height: 6px;
+  width: 12px;
+  left: 4px;
```
Comment (dmcalester) [inline src/css/core/checkbox.css:104]: please change this to 3px 

left: 3px;
Screen Shot 2021-03-02 at 9 06 50 AM

left: 4px;
Screen Shot 2021-03-02 at 9 06 41 AM

**Review by dmcalester — CHANGES_REQUESTED** [review]
One minor change.

**Review by dmcalester — APPROVED** [review]

**Comment by amalyah** [comment]
Could you update the description to have the Jira link that covers checkboxes instead of radio buttons?
https://rocketcom.atlassian.net/browse/ASTRO-1160

File: src/css/core/checkbox.css [hunk 4]
```diff
@@ -0,0 +1,123 @@
+/*
+**
+** 	ASTRO CHECKBOX BUTTON
+** 	==========================================================================
+**  3.0 Notes
+**  Added Custom CSS Properties to support light/dark theming
```
Comment (amalyah) [inline src/css/core/checkbox.css:6]: Missing dashes for bullet points in comment text (if it matters).

File: src/css/core/checkbox.css [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+**  - Removed user-select and placed it in astro.css to apply to all input types
+*/
+
+.rux-checkbox {
+  display: flex;
+  position: relative;
+  margin: 0 0 1rem 0;
+  line-height: 1.2;
+
+  /* DEBUG */
```
Comment (amalyah) [inline src/css/core/checkbox.css:25]: Does this debug code that is  commented out be removed on publishing? If not, we may want to consider deleting this.

File: src/css/core/checkbox.css [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+.rux-checkbox input[type="checkbox"] + label {
+  position: relative;
+  display: flex;
+
+  align-items: center;
+  justify-content: flex-start;
+
+  color: var(--controlLabelColor);
+  letter-spacing: 0.5px;
```
Comment (amalyah) [inline src/css/core/checkbox.css:52]: Does this need to be controlled here? This is the same letter spacing for any other Body 1 text.

File: src/css/core/checkbox.css [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+/* Checkmark */
+.rux-checkbox input[type="checkbox"]:checked + label::after {
+  position: absolute;
+  top: 5px;
+  display: flex;
+  content: "";
+}
+
+.rux-checkbox input[type="checkbox"]:checked + label::after{
+  height: 6px;
```
Comment (amalyah) [inline src/css/core/checkbox.css:102]: What part of the design do these properties correspond to?

File: stories/rux-form-elements.stories.js [hunk 8]
```diff
@@ -45,7 +75,7 @@ Checkboxes.story = {
 };
 
 export const RadioButtons = () => html`
-    <div style="padding: 10vh 5vw; display: flex; justify-content: center;">
+    <div style="padding: 10vh 5vw; display: flex; justify-content: center;" id="hey">
```
Comment (amalyah) [inline stories/rux-form-elements.stories.js:78]: Is there a reason to call this id "hey"?

**Review by amalyah — CHANGES_REQUESTED** [review]
- See comments.
- Also the indeterminate dash part of the checkbox does not appear to be horizontally centered when viewed in Chrome.
- Still not seeing a grouped checkbox sample. Has this feature been cut from the PI?

File: src/css/core/checkbox.css (hunk 7 above)
Comment (Cheshire89) [reply src/css/core/checkbox.css:102]: This is check mark

File: src/css/core/checkbox.css (hunk 6 above)
Comment (Cheshire89) [reply src/css/core/checkbox.css:52]: Not necessarily. However when checking global styles I could not find this property being set globally. Therefore I've resorted to setting it specifically on this element

File: src/css/core/checkbox.css (hunk 5 above)
Comment (Cheshire89) [reply src/css/core/checkbox.css:25]: Agreed. Generally a bad practice keeping debug code around

File: src/css/core/checkbox.css (hunk 4 above)
Comment (Cheshire89) [reply src/css/core/checkbox.css:6]: It doesn't but I'll add them for appearance consistency

File: stories/rux-form-elements.stories.js (hunk 8 above)
Comment (Cheshire89) [reply stories/rux-form-elements.stories.js]: my bad testing code

File: src/css/core/checkbox.css (hunk 7 above)
Comment (amalyah) [reply src/css/core/checkbox.css:89]: Thanks!

File: src/css/core/checkbox.css (hunk 6 above)
Comment (amalyah) [reply src/css/core/checkbox.css:39]: @dmcalester Do you know where/if the letter spacing is being set for Body 1 text throughout the CSS?

File: src/css/core/checkbox.css [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+  width: 12px;
+  left: 3px;
+  border-right: 2px solid var(--controlTextColor);
+  border-top: 2px solid var(--controlTextColor);
+  transform: rotate(125deg);
+}
+
+.rux-checkbox--indeterminate input[type="checkbox"]:checked + label::after{
```
Comment (amalyah) [inline src/css/core/checkbox.css:97]: The indeterminate state dash in the middle of the square is not centered horizontally when viewing in Chrome.

**Review by amalyah — CHANGES_REQUESTED** [review]
1. Is this PR for checkboxes only or also for Radio Buttons and other edits?
Checkbox Jira link that is missing from the description (please add it): https://rocketcom.atlassian.net/browse/ASTRO-11601. 
2. Still not seeing a grouped checkbox sample. Has this feature been cut from the PI on the Storybook side?

File: src/css/core/checkbox.css (hunk 9 above)
Comment (Cheshire89) [reply src/css/core/checkbox.css:97]: My apologies. Once fix broke this one

File: src/css/core/checkbox.css (hunk 6 above)
Comment (Cheshire89) [reply src/css/core/checkbox.css:39]: normally global styles like those are either set via `body` or `*`

File: src/css/core/checkbox.css (hunk 9 above)
Comment (Cheshire89) [reply src/css/core/checkbox.css:97]: The grouped checkbox sample is not feasible due to limitations of storybook and lit-elements working together. I've talked to @dmcalester before dropping attempts of trying to create that. 

storybook version update that is planned after PI release could solve some of those issues but i'm not 100% sure

**Comment by Cheshire89** [comment]
Use react (for display only) to build checkbox group integration on forms -> checkbox page
https://rocketcom.atlassian.net/browse/ASTRO-1272

**Review by cskerbo — APPROVED** [review]

**Review by amalyah — APPROVED** [review]
Looks good. Is there a reason that some pages have a Revision History section but not others? (like this one)

**Pull request merged**